### PR TITLE
fix src/bdict.cc: dict keys should be sorted before encode

### PR DESCRIPTION
**Pull request by ttys3**
this may fixup issue: https://github.com/Frederick888/php-bencode/issues/7

### CODE REVIEW

File: src/zend_container.h [hunk 1]
```diff
@@ -3,6 +3,34 @@
 
 extern "C" {
 #include "php.h"
+
+    //c func borrowed from https://github.com/php/php-src/blob/e08ce4c13db6e9aecd3497cd270b72d06c649bc7/ext/standard/array.c#L245
```
Comment (Frederick888) [inline src/zend_container.h]: Could you please add a space after `//` and format this file using clang-format? There is already a `.clang-format` in the project.

**Comment by Frederick888** [comment]
Ah, good spot. I actually didn't know the fact that 'there is only a single valid bencoding' and sorting keys of dictionaries is apparently needed for this.

There is a minor formatting issue which I already pointed out above and it'd be nice if you could fix it and squash the commits. Thanks for the contribution.

File: src/zend_container.h (hunk 1 above)
Comment (ttys3) [reply src/zend_container.h]: done. force pushed

**Comment by ttys3** [comment]
Yes, I didn't encounter any problems until one day , some users of [Transmission](https://github.com/transmission/transmission) reported to me that the torrent file downloaded can not be used because the private tracker server responsed with : torrent not registered on the server.

then, I did some investigation and found that it is that the php extension we used which generated the wrong bittorrent file.

this `invalid bittorrent file` actually can used with other torrent clients like qBittorrent or Deluge, because when calculating the info hash value, both  qBittorrent and Deluge will respect the key orders in the torrent file. But Transmission client will always re-sort the keys in order, which caused the problem.
And, as a side effect, finally I found the bug in the php-bencode extension and fixed it.

**Comment by ttys3** [comment]
test added

**Comment by Frederick888** [comment]
Thank you. I'll release a new version asap.

**Comment by Frederick888** [comment]
There you go https://github.com/Frederick888/php-bencode/releases/tag/v1.2.5 :)

**Comment by ttys3** [comment]
thanks for the work. `v1.2.5` look really awesome

**Pull request merged**